yoga/algorithm/CalculateLayout.cpp

```diff
@@ -2021,6 +2021,18 @@ static void calculateLayoutImpl(
             currentLead += remainingAlignContentDim / 2;
           }
           break;
+        case Align::SpaceEvenly:
+          if (availableInnerCrossDim > totalLineCrossDim) {
+            currentLead +=
+                remainingAlignContentDim / static_cast<float>(lineCount + 1);
+            if (lineCount > 1) {
+              crossDimLead =
+                  remainingAlignContentDim / static_cast<float>(lineCount + 1);
+            }
+          } else {
+            currentLead += remainingAlignContentDim / 2;
+          }
+          break;
```

Comment:
Am I following this logic correctly?

`currentLead` is the leading position of the current line (incremented by each line height), then `crossDimLead` is the per-line lead added to each line after.

When there is not free space left, we fall back to centering. This is the specified behavior in the [flexbox spec](https://www.w3.org/TR/css-flexbox-1/#valdef-align-content-space-around) for `space-around`, where the later [box alignment](https://www.w3.org/TR/css-align-3/#distribution-values) spec instead says `space-around` and `space-evenly` have a fallback alignment of "safe center"?

Reply:
> e.g. I think https://github.com/facebook/yoga/pull/1013 probably still needs to happen)

Yuck, I see. We are contributing `crossDimLead` to not just distribution of the line, but then also using it for alignment within the line.

Is there a reason for `if (lineCount > 1)` here? Or was it just around for `Align::SpaceAround` to avoid division by zero, if the lead will never be used after the next line (ignoring the bug where it contributes to `align-items`).

Reply:
> currentLead is the leading position of the current line (incremented by each line height), then crossDimLead is the per-line lead added to each line after.

I think that's correct. Although, I believe the effect of the code adding to `currentLead` here is to set the "lead" for the *first* line.

> Is there a reason for if (lineCount > 1) here?

I don't think so. I just left it there on a principle of not wanting to touch anything. I think you're probably right about it being there to avoid a divide by zero. Although it seems like it might be better not to run alignment at all if there are zero lines (Taffy actually skips the entire Flexbox algorithm if there are no children as it reduces to relatively trivial calculations involving the container node's styles)

> When there is not free space left, we fall back to centering. 

Yep

> This is the specified behavior in the [flexbox spec](https://www.w3.org/TR/css-flexbox-1/#valdef-align-content-space-around) for space-around, where the later [box alignment](https://www.w3.org/TR/css-align-3/#distribution-values) spec instead says space-around and space-evenly have a fallback alignment of "safe center"?

I believe the "safe" alignment modes are opt-in. So these would probably be separate enum variants (or a boolean on top of the alignment) in Yoga.